src/SettingsManager.js

```diff
@@ -0,0 +1,122 @@
+'use strict'
+
+let equal = require('deep-equal');
+let deepcopy = require('deepcopy');
+
+import { logger, configureLogger } from './logger';
+import cache from './cache';
+import Config from './Config';
+let authDataManager = require('./authDataManager');
+
+
+const settingsCollectionName = '_ServerSettings';
+
+// visible locked settings
+const lockedSettings = [
+  'applicationId',
+  'masterKey',
+  'serverURL',
+  'collectionPrefix',
+  'enableConfigChanges',
+];
+// hidden locked settings
+const hiddenSettings = [
+  'filesController',
+  'pushController',
+  'loggerController',
+  'hooksController',
+  'userController',
+  'authDataManager',
+  'liveQueryController',
+  'settingsInitialized'
+];
+
+// settings defined in server configuration
+let definedSettings = {};
+
+// callbacks for specific setting changes
+let authChange = settings => settings.authDataManager = authDataManager(settings.ouath, settings.enableAnonymousUsers);
+let onChange = {
+  logLevel: settings => configureLogger({ level: settings.logLevel }),
+  verbose: settings => {
+    settings.logLevel = 'silly';
+    configureLogger({ level: settings.logLevel });
```

Comment:
Not a fan of having both logLevel & verbose settings as the pair leads to weird interactions, but not sure how better to handle it